internal/provider/validators_test.go

```diff
@@ -261,3 +261,83 @@ func Test_validateDatacenter(t *testing.T) {
 		})
 	}
 }
+
+func Test_validateConsulClusterTier(t *testing.T) {
+	tcs := map[string]struct {
+		input    string
+		expected diag.Diagnostics
+	}{
+		"valid tier lowercase": {
+			input:    "development",
+			expected: nil,
+		},
+		"valid tier uppercase": {
+			input:    "STANDARD",
+			expected: nil,
+		},
+		"valid tier mixedcase": {
+			input:    "DEVelopment",
+			expected: nil,
+		},
+		"invalid tier": {
+			input: "dev",
+			expected: diag.Diagnostics{
+				diag.Diagnostic{
+					Severity:      diag.Error,
+					Summary:       "expected dev to be one of [UNSET DEVELOPMENT STANDARD]",
```

Comment:
this is interesting because tier=UNSET will fail in Consul Service since it is validated as required. I like the validation functions you added but is there an easy way to remove UNSET as being suggested? (same comment about the VM size validation).

Reply:
RE "same comment about the VM size validation" above. I thought more about it and this isn't the same for VM size since it's optional. But still seems weird to be suggesting UNSET as an option.

Reply:
Yea UNSET is a weird option for sure. I am guessing it originates from the proto file having it [here](https://github.com/hashicorp/cloud-consul-service/blob/master/proto-public/20210204/config.proto#L32) and gets propogated in by the swagger tools. 
Wondering instead of doing any manipulation here, we can avoid generating it in the [SDK](https://github.com/hashicorp/hcp-sdk-go/blob/main/clients/cloud-consul-service/preview/2021-02-04/models/hashicorp_cloud_consul20210204_cluster_config_tier.go#L24) itself 🤔

Reply:
@bcmdarroch @roaks3 any thoughts ^^? 
That said, I am happy to add some manipulation to remove the UNSET option from the suggested list (might be easier in the short term too). Let me know.

Reply:
Yeah I think in the short-term it'd be best to remove it here. Long-term, I'm thinking of updating the SDK template to provide a constant that's an array of all possible enum values, and we could drop the UNSET there. Then that constant could be used in the validator function, in the documentation, etc.

Reply:
Great! Thanks @bcmdarroch!

Reply:
Removed the UNSET option for the short term.